client/src/components/AssignPersonModal.js

```diff
@@ -124,90 +136,139 @@ const AssignPersonModal = ({ position, showModal, onCancel, onSuccess }) => {
-                  <td>{person.name}</td>
-                  <td>
-                    {person.position ? (
-                      person.position.name
-                    ) : person.uuid === position.person.uuid ? (
-                      position.name
-                    ) : (
-                      <i>None</i>
-                    )}
-                  </td>
-                </tr>
-              </tbody>
-            </Table>
-          )}
-          <Messages error={error} />
-        </Grid>
+        {!removeUser && (
+          <Grid fluid>
+            <Row>
+              <Col md={12}>
+                <FormGroup controlId="person" required>
+                  <AdvancedSingleSelect
+                    fieldName="person"
+                    fieldLabel="Select a person"
+                    placeholder="Select a person for this position"
+                    value={person}
+                    overlayColumns={["Name"]}
+                    overlayRenderRow={PersonSimpleOverlayRow}
+                    filterDefs={personFilters}
+                    onChange={value => setPerson(value)}
+                    objectType={Person}
+                    valueKey="name"
+                    fields="uuid, name, rank, role, avatar(size: 32), position { uuid, name, type, organization {uuid} }"
+                    addon={PEOPLE_ICON}
+                    vertical
+                  />
+                </FormGroup>
+              </Col>
+            </Row>
+            {person && person.uuid && (
+              <Table striped condensed hover responsive>
+                <thead>
+                  <tr>
+                    <th>Rank</th>
+                    <th>Name</th>
+                    <th>Current Position</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <tr>
+                    <td>{person.rank}</td>
+                    <td>{person.name}</td>
+                    <td>
+                      {person.position ? (
+                        person.position.name
+                      ) : person.uuid === position.person.uuid ? (
+                        position.name
+                      ) : (
+                        <i>None</i>
+                      )}
+                    </td>
+                  </tr>
+                </tbody>
+              </Table>
+            )}
+            {<Messages error={error} />}
+          </Grid>
+        )}
       </Modal.Body>
       <Modal.Footer>
-        <Button className="pull-left" onClick={closeModal}>
+        <Button
+          className="pull-left"
+          onClick={() => {
+            removeUser ? setRemoveUser(false) : closeModal()
+          }}
+        >
           Cancel
         </Button>
-        <Button onClick={save} bsStyle="primary" className="save-button">
+        <Button
+          onClick={() => {
+            if (removeUser) {
+              setPerson(null)
+              setDoSave(true)
+              setRemoveUser(false)
```

Comment:
`setRemoveUser(false)` is executed under all branch conditions, so move it outside the `if`.